pkg/api/validate/dynamic/dynamic.go

```diff
@@ -32,6 +32,12 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/token"
 )
 
+var (
+	errMsgNSGAttached            = "The provided subnet '%s' is invalid: must not have a network security group attached."
+	errMsgNSGNotAttached         = "The provided subnet '%s' is invalid: must have network security group '%s' attached."
+	errMsgNSGNotProperlyAttached = "BYO NSG mode requires that all subnets are attached with an NSG."
```

Comment:
Assuming Puneet is good with these error messages.  if not please check with him :)